DUI3-DX/Connectors/Autocad/Speckle.Connectors.AutocadShared/Bindings/AutocadReceiveBinding.cs

```diff
@@ -0,0 +1,231 @@
+      {
+        objectsToConvert.Add((objPath, obj));
+      }
+    }
+
+    return objectsToConvert;
+  }
+
+  private List<string> ConvertObjects(Base commitObject, ReceiverModelCard modelCard, CancellationTokenSource cts)
+  {
+    // Prompt the UI conversion started. Progress bar will swoosh.
+    Commands.SetModelProgress(modelCard.ModelCardId, new ModelCardProgress() { Status = "Converting" });
+
+    ISpeckleConverterToHost converter = _speckleConverterToHostFactory.ResolveScopedInstance();
+
+    // Layer filter for received commit with project and model name
+    _autocadLayerManager.CreateLayerFilter(modelCard.ProjectName, modelCard.ModelName);
+    List<(List<string>, Base)> objectsWithPath = GetBaseWithPath(commitObject, cts);
+    string baseLayerPrefix = $"SPK-{modelCard.ProjectName}-{modelCard.ModelName}-";
+
+    HashSet<string> uniqueLayerNames = new();
+    List<string> handleValues = new();
+    int count = 0;
+
+    using (TransactionContext.StartTransaction(Application.DocumentManager.MdiActiveDocument))
+    {
+      foreach ((List<string> path, Base obj) in objectsWithPath)
+      {
+        if (cts.IsCancellationRequested)
+        {
+          throw new OperationCanceledException(cts.Token);
+        }
+
+        try
+        {
+          string layerFullName = _autocadLayerManager.LayerFullName(baseLayerPrefix, string.Join("-", path));
+
+          if (uniqueLayerNames.Add(layerFullName))
+          {
+            _autocadLayerManager.CreateLayerOrPurge(layerFullName);
+          }
+
+          object converted = converter.Convert(obj);
+          List<object> flattened = FlattenToNativeConversionResult(converted);
+
+          foreach (Entity conversionResult in flattened.Cast<Entity>())
+          {
+            if (conversionResult == null)
+            {
+              continue;
+            }
+
+            conversionResult.Append(layerFullName);
+            handleValues.Add(conversionResult.Handle.Value.ToString());
+          }
+
+          Commands.SetModelProgress(
+            modelCard.ModelCardId,
+            new ModelCardProgress() { Status = "Converting", Progress = (double)++count / objectsWithPath.Count }
+          );
+        }
+        catch (Exception e) when (!e.IsFatal()) // DO NOT CATCH SPECIFIC STUFF, conversion errors should be recoverable
+        {
+          // POC: report, etc.
+          Debug.WriteLine("conversion error happened.");
+        }
+      }
+    }
+    return handleValues;
+  }
+
+  // POC: This method should be move somewhere more general!
+  /// <summary>
+  /// Utility function to flatten a conversion result that might have nested lists of objects.
+  /// This happens, for example, in the case of multiple display value fallbacks for a given object.
+  /// </summary>
+  /// <param name="item"></param>
+  /// <returns></returns>
+  private List<object> FlattenToNativeConversionResult(object item)
+  {
+    List<object> convertedList = new();
+    void Flatten(object item)
```

Comment:
private local functions are somewhat controversial. I don't see why this needs to be a private function. It makes the logic a bit convoluted (and recursive too 😅)

We could flatten the recursive pattern into a simpler implementation using a `while` loop, which would also remove the need for the private function altogether.

Reply:
Agree, added also into Core Utilities and unit test for new implementation.